internal/qkcapi/api.go

```diff
@@ -64,31 +61,64 @@ func (c *CommonAPI) SendRawTransaction(encodedTx hexutil.Bytes) (hexutil.Bytes,
 		return EmptyTxID, err
 	}
 	return encoder.IDEncoder(tx.Hash().Bytes(), tx.EvmTx.FromFullShardKey()), nil
+
 }
 
-func (c *CommonAPI) GetTransactionReceipt(txID hexutil.Bytes) (map[string]interface{}, error) {
-	txHash, fullShardKey, err := encoder.IDDecoder(txID)
-	if err != nil {
-		return nil, err
+func MetaMaskReceipt(block *types.MinorBlock, i int, receipt *types.Receipt) (map[string]interface{}, error) {
+	if block == nil {
+		return nil, errors.New("block is nil")
 	}
-
-	fullShardId, err := clusterCfg.Quarkchain.GetFullShardIdByFullShardKey(fullShardKey)
+	txHash := ""
+	if len(block.Transactions()) > i {
+		tx := block.Transactions()[i]
+		txHash = tx.Hash().String()
+	}
+	if receipt == nil {
+		return nil, errors.New("receipt is nil")
+	}
+	header := block.Header()
+	tx := block.Transactions()[i]
+	from, err := tx.Sender(types.NewEIP155Signer(clusterCfg.Quarkchain.NetworkID))
 	if err != nil {
 		return nil, err
 	}
-	branch := account.Branch{Value: fullShardId}
-	minorBlock, index, receipt, err := c.b.GetTransactionReceipt(txHash, branch)
+	field := map[string]interface{}{
```

Comment:
will this change impact the current service, does python need to change also?

Reply:
will revert